homeassistant/components/home_connect/coordinator.py

```diff
@@ -0,0 +1,227 @@
+        """Match event with listener for event type."""
+        while True:
+            try:
+                async for event_message in self.client.stream_all_events():
+                    match event_message.type:
+                        case EventType.STATUS:
+                            statuses = self.data[event_message.ha_id].status
+                            for event in event_message.data.items:
+                                if (
+                                    event.key not in StatusKey
+                                    or event.key is EventKey.UNKNOWN
+                                ):
+                                    continue
+                                status_key = StatusKey(event.key)
+                                if status_key in statuses:
+                                    statuses[status_key].value = event.value
+                                else:
+                                    statuses[status_key] = Status(
+                                        status_key, event.value
+                                    )
+                            await self._call_event_listener(event_message)
+
+                        case EventType.NOTIFY:
+                            settings = self.data[event_message.ha_id].settings
+                            for event in event_message.data.items:
+                                if (
+                                    event.key not in SettingKey
+                                    or event.key is EventKey.UNKNOWN
+                                ):
+                                    continue
+                                setting_key = SettingKey(event.key)
+                                if setting_key in settings:
+                                    settings[setting_key].value = event.value
+                                else:
+                                    settings[setting_key] = GetSetting(
+                                        setting_key, event.value
+                                    )
+                            await self._call_event_listener(event_message)
+
+                        case EventType.EVENT:
+                            await self._call_event_listener(event_message)
+
+            except (EventStreamInterruptedError, HomeConnectRequestError) as error:
+                _LOGGER.debug(
+                    "Non-breaking error (%s) while listening for events,"
+                    " continuing in 30 seconds",
+                    type(error).__name__,
+                )
+                await asyncio.sleep(EVENT_STREAM_RECONNECT_DELAY)
+            except HomeConnectApiError as error:
+                _LOGGER.error("Error while listening for events: %s", error)
+                break
+
+    async def _call_event_listener(self, event_message: EventMessage):
+        """Call listener for event."""
+        listeners = self._home_appliances_event_listeners.get(event_message.ha_id, {})
```

Comment:
I'd look into using the coordinator context to selectively update the correct entities. The entities will already be connected to the coordinator if they inherit `CoordinatorEntity`.

We should update the coordinator state here in the coordinator from the event and then let the correct entities update their state.

Reply:
Here's a example how to use the context:

https://github.com/home-assistant/core/blob/b8632063f5d86931393abdc9840a41831a211470/homeassistant/components/nibe_heatpump/coordinator.py#L25-L63